syntax = "proto3";
package idcrack;




import "idcmanager_sdk/model/idcmanager/idc.proto";
import "idcmanager_sdk/model/idcmanager/host.proto";
import "idcmanager_sdk/model/idcmanager/switch.proto";
import "idcmanager_sdk/model/idcmanager/router.proto";
import "idcmanager_sdk/model/idcmanager/firewall.proto";

/*
ListIDCRack请求
*/
message ListIDCRackRequest {
    /*
    机柜ID列表，以,分割
    */
    string idcrackIds = 1 ;
}

/*
ListIDCRack返回
*/
message ListIDCRackResponse {
message List {
message Layout {
    /*
    设备ID
    */
    string instanceId = 1 ;
    /*
    设备类型(host,switch,router,firewall,disable)
    */
    string type = 2 ;
    /*
    起始U位
    */
    int32 startU = 3 ;
    /*
    占用U位
    */
    int32 occupiedU = 4 ;
}
        
    
    /*
    所在的机房
    */
    idcmanager.IDC idc = 1 ;
    /*
    机柜上的主机
    */
    repeated idcmanager.Host hosts = 2 ;
    /*
    机柜上的交换机
    */
    repeated idcmanager.Switch switchs = 3 ;
    /*
    机柜上的路由器
    */
    repeated idcmanager.Router routers = 4 ;
    /*
    机柜上的防火墙
    */
    repeated idcmanager.Firewall firewalls = 5 ;
    /*
    机柜ID
    */
    string instanceId = 6 ;
    /*
    机柜名称
    */
    string name = 7 ;
    /*
    机柜编号
    */
    string code = 8 ;
    /*
    机柜状态
    */
    string status = 9 ;
    /*
    机柜类型
    */
    string type = 10 ;
    /*
    机柜U数
    */
    int32 unum = 11 ;
    /*
    空闲U位数
    */
    int32 freeUnum = 12 ;
    /*
    备注
    */
    string memo = 13 ;
    /*
    机柜部局
    */
    repeated Layout layout = 14 ;
    /*
    创建时间
    */
    string ctime = 15 ;
    /*
    创建者
    */
    string creator = 16 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListIDCRackApi返回
*/
message ListIDCRackResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    idcrack.ListIDCRackResponse data = 4 ;
}

